src/common.nugetversions.targets

```diff
@@ -0,0 +1,97 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <!--===============================================================================================================
+    SPECIAL SAUCE
+    ===================================================================================================================  
+    This section contains any special sauce needed to implement Global NuGet Management.-->
+
+  <!--Since Global NuGet Management requires having a single [packages.config] file that's located in a special
+        global location, its existence won't be known by default to Projects when they are opened/built.  As such,
+        we need the following target that runs before Project References are resolved, which is responsible for
+        downloading and using [NuGet.exe] to perform a manual restore operation on the global [packages.config]
+        file.  We run this target only for local developer inner loop scenarios (BuildingInsideVisualStudio=true).  For
+        Azure DevOps builds, a NuGet Restore pipeline task is run against the [packages.config] file instead.-->
+
+  <PropertyGroup>
+    <Terminal_NuGetEXE>$(SolutionDir)dep\nuget\nuget.exe</Terminal_NuGetEXE>
+    <Terminal_GlobalPackagesConfigPath>$(SolutionDir)dep\nuget\packages.config</Terminal_GlobalPackagesConfigPath>
+  </PropertyGroup>
+
+  <Target Name="_Terminal_NuGetPackageManager"
+      BeforeTargets="ResolveProjectReferences"
+      Inputs="$(Terminal_GlobalPackagesConfigPath)"
+      Outputs="$(SolutionDir)packages\packages.config"
+      Condition="'$(BuildingInsideVisualStudio)' == 'true'">
+
+    <!--Have the [nuget.exe] tool do a restore on the global [packages.config] file for the repository.-->
+    <Exec Command="$(Terminal_NuGetEXE) install $(Terminal_GlobalPackagesConfigPath)" />

```

Comment:
I have a vague lingering horror about starting from a clean build and running a `/m` build... Once-per-project non-global-locked `Exec` tasks have deeply hurt me in the past . . . so how do we make sure this only happens ONCE?

I chose to handle this for the feature staging header with a separate `.proj`:

https://github.com/microsoft/terminal/blob/830a422c015a64cb37196c0b55c2d35d79692fc3/src/common.build.post.props#L66-L75

https://github.com/microsoft/terminal/blob/main/build/rules/GenerateFeatureFlags.proj

Reply:
So the concern is that if there is a parallel build and the dependencies are not already downloaded then there could be two instances of nuget.exe at the same time stepping on each other?  That does seem like a plausible risk although I did not see it first-hand because the restore happens when the solution is loaded so it is usually (always?) done before I can build for the first time.